### PR TITLE
pkg/nimble/scanner: add function to set scan duration 

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adds a function to enable setting the scan duration, similar to what is done for the autoadv module. By default it will run forever. The `nimble_scanner` example has been adapted accordingly.

### Testing procedure

`make -C examples/nimble_scanner/ flash term`

```
2021-08-03 12:00:47,058 # main(): This is RIOT! (Version: 2021.10-devel-270-g0202e2-pr_nimble_scanner_set_duration)
2021-08-03 12:00:47,059 # NimBLE Scanner Example Application
2021-08-03 12:00:47,059 # Type `scan help` for more information
> scan 1000
2021-08-03 12:01:21,922 # scan 1000
2021-08-03 12:01:22,925 # Scanning for 1000 ms now ... done
2021-08-03 12:01:22,926 # 
2021-08-03 12:01:22,926 # Results:
2021-08-03 12:01:22,935 # [ 0] e4:dd:e0:8f:73:65 (RANDOM) [IND] "RIOT-autoconn", adv_msg_cnt: 19, adv_int: 49093us, last_rssi: -55
2021-08-03 12:01:22,936 # 
> scan
2021-08-03 12:01:24,617 # scan
2021-08-03 12:01:25,620 # Scanning for 1000 ms now ... done
2021-08-03 12:01:25,621 # 
2021-08-03 12:01:25,621 # Results:
2021-08-03 12:01:25,630 # [ 0] e4:dd:e0:8f:73:65 (RANDOM) [IND] "RIOT-autoconn", adv_msg_cnt: 18, adv_int: 47056us, last_rssi: -55
2021-08-03 12:01:25,630 # 
```

### CODE REVIEW

File: examples/nimble_scanner/main.c [hunk 1]
```diff
@@ -22,33 +22,34 @@
...
-    uint32_t timeout = DEFAULT_DURATION;
+    uint32_t timeout = DEFAULT_DURATION_MS;
 
     if ((argc == 2) && (memcmp(argv[1], "help", 4) == 0)) {
         printf("usage: %s [timeout in ms]\n", argv[0]);
         return 0;
     }
     if (argc >= 2) {
-        timeout = (uint32_t)(atoi(argv[1]) * 1000);
+        timeout = (uint32_t)atoi(argv[1]);
```
Comment (benpicco) [inline examples/nimble_scanner/main.c:45]:
```suggestion
        timeout = atoi(argv[1]);
```

File: pkg/nimble/scanner/include/nimble_scanner.h [hunk 2]
```diff
@@ -68,6 +68,9 @@ int nimble_scanner_init(const struct ble_gap_disc_params *params,
 
 /**
  * @brief   Start scanning using timing parameters configured on initialization
+ *
+ * @note    Scanning will run for ever unless stopped or unless a different
+ *          scan duration is set with @ref nimble_scan_set_scan_duration
```
Comment (benpicco) [inline pkg/nimble/scanner/include/nimble_scanner.h:73]: `nimble_scan_set_scan_duration` does not exist in doxygen

File: pkg/nimble/scanner/nimble_scanner.c [hunk 3]
```diff
@@ -32,13 +32,19 @@
 static nimble_scanner_cb _disc_cb = NULL;
 static struct ble_gap_disc_params _scan_params = { 0 };
 
+/* duration of the scanning procedure */
+static int32_t _scan_duration;
```
Comment (benpicco) [inline pkg/nimble/scanner/nimble_scanner.c:36]: Why not
```suggestion
static int32_t _scan_duration = BLE_HS_FOREVER;
```

File: pkg/nimble/scanner/nimble_scanner.c [hunk 4]
```diff
@@ -92,5 +107,7 @@ int nimble_scanner_init(const struct ble_gap_disc_params *params,
     }
     _disc_cb = disc_cb;
 
+    _scan_duration = BLE_HS_FOREVER;
+
```
Comment (benpicco) [inline pkg/nimble/scanner/nimble_scanner.c:111]:
```suggestion
```

File: pkg/nimble/scanner/nimble_scanner.c (hunk 3 above)
Comment (fjmolinas) [reply pkg/nimble/scanner/nimble_scanner.c]: addressed

File: pkg/nimble/scanner/nimble_scanner.c (hunk 4 above)
Comment (fjmolinas) [reply pkg/nimble/scanner/nimble_scanner.c]: addressed

File: examples/nimble_scanner/main.c (hunk 1 above)
Comment (fjmolinas) [reply examples/nimble_scanner/main.c]: addressed

File: pkg/nimble/scanner/include/nimble_scanner.h (hunk 2 above)
Comment (fjmolinas) [reply pkg/nimble/scanner/include/nimble_scanner.h]: addressed

**Comment by fjmolinas** [comment]
May I squash @benpicco?

**Comment by benpicco** [comment]
sure, please squash!

**Comment by benpicco** [comment]
Just wondering: Adding a `scan_duration` parameter to `nimble_scanner_start()` was not an option?

**Comment by fjmolinas** [comment]
> Just wondering: Adding a `scan_duration` parameter to `nimble_scanner_start()` was not an option?

I think it was initially done to mimic the `auto_adv` API, and since there is a start/stop you can change it..

**Review by benpicco — APPROVED** [review]

**Comment by fjmolinas** [comment]
Thanks for the review!

**Pull request merged**